File: app/src/main/java/com/example/newyorktimearticles/data/repository/GetArticlesRepository.java
package com.example.newyorktimearticles.data.repository;

import com.example.newyorktimearticles.data.model.Result;
import com.example.newyorktimearticles.data.webservice.ApiService;
import com.example.newyorktimearticles.data.webservice.Constants;

import io.reactivex.Single;
import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;

public class GetArticlesRepository implements IGetArticlesRepository {
    ApiService apiService;

    public GetArticlesRepository() {
        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl(Constants.baseUrl)
                .addConverterFactory(GsonConverterFactory.create())
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .client(getClient())
                .build();
        apiService = retrofit.create(ApiService.class);
    }

    public OkHttpClient getClient() {
        HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor();
        interceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
        return new OkHttpClient.Builder().addInterceptor(interceptor).build();
    }

    @Override
    public Single<Result> getArticleList() {
        return apiService.getPopularArticles(Constants.apiKey);
    }
}

File: app/src/main/java/com/gemidroid/newyorktimearticles/data/repository/GetArticlesRepository.kt
package com.gemidroid.newyorktimearticles.data.repository

import com.gemidroid.newyorktimearticles.data.model.Result
import com.gemidroid.newyorktimearticles.data.webservice.ApiService
import com.gemidroid.newyorktimearticles.utility.Constants
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.flowOf

class GetArticlesRepository(private var apiService: ApiService) : IGetArticlesRepository {

    override suspend fun getArticleList(): Flow<Result> {
        return flowOf(apiService.getPopularArticles(Constants.apiKey))
    }
}